src/Libraries/CoreNodeModels/Logic/If.cs

```diff
@@ -72,4 +78,44 @@ public override IEnumerable<AssociativeNode> BuildOutputAst(List<AssociativeNode
             };
         }
     }
+
+    [NodeName("If")]
+    [NodeCategory(BuiltinNodeCategories.LOGIC)]
+    [NodeDescription("ScopeIfDescription", typeof(Resources))]
+    [OutPortTypes("Function")]
+    [IsDesignScriptCompatible]
+    [AlsoKnownAs("DSCoreNodesUI.Logic.If")]
+    public class RefactoredIf : NodeModel
+    {
+        [JsonConstructor]
+        private RefactoredIf(IEnumerable<PortModel> inPorts, IEnumerable<PortModel> outPorts) : base(inPorts, outPorts) { }
+
+        public RefactoredIf()
+        {
+            InPorts.Add(new PortModel(PortType.Input, this, new PortData("test", Resources.PortDataTestBlockToolTip)));
+            InPorts.Add(new PortModel(PortType.Input, this, new PortData("true", Resources.PortDataTrueBlockToolTip)));
+            InPorts.Add(new PortModel(PortType.Input, this, new PortData("false", Resources.PortDataFalseBlockToolTip)));
+
+            OutPorts.Add(new PortModel(PortType.Output, this, new PortData("result", Resources.PortDataResultToolTip)));
+
+            RegisterAllPorts();
+        }
+
+        public override IEnumerable<AssociativeNode> BuildOutputAst(List<AssociativeNode> inputAstNodes)
+        {
+            var lhs = GetAstIdentifierForOutputIndex(0);
+            AssociativeNode rhs;
+
+            var conditional = AstFactory.BuildConditionalNode(inputAstNodes[0], new IntNode(0), new IntNode(1));
+            var trueStmt = inputAstNodes[1];
+            var falseStmt = inputAstNodes[2];
+            var list = AstFactory.BuildExprList(new List<AssociativeNode> { trueStmt, falseStmt });
+            rhs = AstFactory.BuildIndexExpression(list, conditional);
```

Comment:
Would you add some comments to these lines indicating the difference with the old node?

Reply:
Added it.

Reply:
@reddyashish Awesome. :shipit: